File: ts/src/events.ts
import type {
    ActionExecutionRequest,
    ActionFlow,
    ActionFlowExecutionResponse,
    ActionSubFlowExecutionResponse,
    ActionTransferRequest,
    ActionTransferResponse
} from "@code0-tech/tucana/aquila";
import type {ModuleConfigurations} from "@code0-tech/tucana/shared";
import type {Action} from "./action.ts";

export enum CodeZeroEvent {
    error = "error",
    connected = "connected",
    streamMessageReceived = "streamMessageReceived",
    streamMessageSent = "streamMessageSent",
    moduleUpdated = "moduleUpdated",
    executionRequestReceived = "executionRequestReceived",
    subFlowExecutionResponseReceived = "subFlowExecutionResponseReceived",
    flowExecutionResponseReceived = "flowExecutionResponseReceived",
    flowUpdated = "flowUpdated",
    flowDeleted = "flowDeleted",
}

export interface CodeZeroEventMap {
    [CodeZeroEvent.error]: [Error]
    [CodeZeroEvent.connected]: [Action]
    [CodeZeroEvent.streamMessageReceived]: [ActionTransferResponse]
    [CodeZeroEvent.streamMessageSent]: [ActionTransferRequest]
    [CodeZeroEvent.moduleUpdated]: [ModuleConfigurations]
    [CodeZeroEvent.executionRequestReceived]: [ActionExecutionRequest]
    [CodeZeroEvent.subFlowExecutionResponseReceived]: [ActionSubFlowExecutionResponse]
    [CodeZeroEvent.flowExecutionResponseReceived]: [ActionFlowExecutionResponse]
    [CodeZeroEvent.flowUpdated]: [ActionFlow]
    [CodeZeroEvent.flowDeleted]: [bigint]
    [key: string]: unknown[]
}

File: py/hercules/events.py
"""Event names (port of ``src/events.ts``)."""
from __future__ import annotations

from enum import Enum


class CodeZeroEvent(str, Enum):
    error = "error"
    connected = "connected"
    stream_message_received = "streamMessageReceived"
    stream_message_sent = "streamMessageSent"
    module_updated = "moduleUpdated"
    execution_request_received = "executionRequestReceived"
    sub_flow_execution_response_received = "subFlowExecutionResponseReceived"
    flow_execution_response_received = "flowExecutionResponseReceived"
    flow_updated = "flowUpdated"
    flow_deleted = "flowDeleted"

    def __str__(self) -> str:  # so it can be used interchangeably with the value
        return self.value


__all__ = ["CodeZeroEvent"]
